src/type/validate.js

```diff
@@ -314,94 +317,110 @@ function validateFields(
   }
 }
 
-function validateObjectInterfaces(
+function validateInterfaces(
   context: SchemaValidationContext,
-  object: GraphQLObjectType,
+  implementing: GraphQLObjectType | GraphQLInterfaceType,
 ): void {
   const implementedTypeNames = Object.create(null);
-  for (const iface of object.getInterfaces()) {
-    if (!isInterfaceType(iface)) {
+  for (const implemented of implementing.getInterfaces()) {
+    if (!isInterfaceType(implemented)) {
       context.reportError(
-        `Type ${inspect(object)} must only implement Interface types, ` +
-          `it cannot implement ${inspect(iface)}.`,
-        getAllImplementsInterfaceNodes(object, iface),
+        `Type ${inspect(implementing)} must only implement Interface types, ` +
+          `it cannot implement ${inspect(implemented)}.`,
+        getAllImplementsInterfaceNodes(implementing, implemented),
       );
       continue;
     }
 
-    if (implementedTypeNames[iface.name]) {
+    if (implementedTypeNames[implemented.name]) {
       context.reportError(
-        `Type ${object.name} can only implement ${iface.name} once.`,
-        getAllImplementsInterfaceNodes(object, iface),
+        `Type ${implementing.name} can only implement ${implemented.name} once.`,
+        getAllImplementsInterfaceNodes(implementing, implemented),
       );
       continue;
     }
-    implementedTypeNames[iface.name] = true;
-    validateObjectImplementsInterface(context, object, iface);
+    implementedTypeNames[implemented.name] = true;
+    validateImplementsInterface(context, implementing, implemented);
   }
 }
 
-function validateObjectImplementsInterface(
+function validateImplementsInterface(
   context: SchemaValidationContext,
-  object: GraphQLObjectType,
-  iface: GraphQLInterfaceType,
+  implementing: GraphQLObjectType | GraphQLInterfaceType,
+  implemented: GraphQLInterfaceType,
```

Comment:
Same here, I think `type` and `iface` is perfectly fine.
This also will minimize diff.
I agree that renaming stuff may be beneficial but it should be done in separate PR and for the entire file so we have consistency.

Reply:
With all renaming going on it's extremely hard to spot actual changes :(

Reply:
Thi follows the language in the spec, which is very helpful when verifying correctness.

Reply:
With that in mind, would you still prefer this be in a separate PR?

Reply:
I've gone ahead and renamed these per your suggestion.

Reply:
So there are two problems here:
1. Since now it validates both interface and objects but in your version, it matches only interface algorithm.
2. Maybe it's only me but `implementing` and `implemented` looks too similar when I'm scanning through the source code. Maybe it's because I'm not a native speaker or because of some mild dyslexia.

Reply:
1. Yes, originally I had two different functions: one for objects and one for interfaces. This allowed me to use language that matched the spec in both places. You're right, though, that since we combined them into one function, only one will have wording that matches the spec.
2. Yes, this is a bit hard to read, but in the context of interfaces implementing other interfaces, these were the most clear words I could think to use. I agree that while that might be a good choice in the spec (for the sake of correctness) it's much easier to follow here, and probably the right choice given point 1.